pyfr/plugins/fluidforce.py

```diff
@@ -122,6 +160,29 @@ def __call__(self, intg):
             # Do the quadrature
             f += np.einsum('i...,ij,jik', qwts, p, norms)
 
+            if self.forcetype == 'viscous':
+                # Get operator and J^-T matrix
+                m4 = self._m4[etype]
+                rcpjact = self._rcpjact[etype, fidx]
+
+                # Transformed gradient at solution points
+                tduupts = np.dot(m4, uupts.reshape(nupts, -1))
+                tduupts = tduupts.reshape(ndims, nupts, nvars, -1)
+
+                # Physical gradient at solution points
+                duupts = np.einsum('ijkl,jkml->ikml', rcpjact, tduupts)
+
+                # Interpolate gradient to flux points
+                m5 = block_diag([m0]*ndims)
+                dufpts = np.dot(m5, duupts.reshape(ndims*nupts, -1))
+                dufpts = dufpts.reshape(ndims, nfpts, nvars, -1).swapaxes(1, 2)
```

Comment:
I have compared these three lines (176~ 178) and multiple application of m0, which is followings.

```python
duupts = duupts.reshape(ndims, nupts, -1)
dufpts = np.zeros((ndims, nfpts) + duupts.shape[2:])
for i in range(ndims):
    dufpts[i] = np.dot(m0, duupts[i])
    
dufpts = dufpts.reshape(ndims, nfpts, nvars, -1).swapaxes(1, 2)
```

For (nfpts, nupts, ndims, nvars, neles) = (16, 64, 3, 5, 1024), current implementation and multiple application of m0 tooks 16.6ns and 20.6ns, respectively. The current one is slightly faster. Do we need to change these lines?